patch/kernel/archive/rockchip64-6.12/board-rockpis-dts-fixes.patch

```diff
@@ -0,0 +1,130 @@
+From 3bdc6856ef82b6619c527034206b49a69f3d1b91 Mon Sep 17 00:00:00 2001
+From: Paolo Sabatino <paolo.sabatino@gmail.com>
+Date: Tue, 26 Nov 2024 12:13:24 +0100
+Subject: [PATCH] rk3308: fixes for rock pi s dts
+
+---
+ .../boot/dts/rockchip/rk3308-rock-pi-s.dts    | 75 +++++++++++++++++++
+ 1 file changed, 75 insertions(+)
+
+diff --git a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts b/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
+index 62d18ca769a1..8bcc57e9440c 100644
+--- a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
++++ b/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts
+@@ -48,6 +48,54 @@ blue-led {
+ 		};
+ 	};
+ 
++	analog_sound: analog-sound {
++		compatible = "simple-audio-card";
++		simple-audio-card,format = "i2s";
++		simple-audio-card,name = "analog";
++		simple-audio-card,mclk-fs = <256>;
++		simple-audio-card,codec-hp-det;
++		simple-audio-card,widgets =
++			"Headphone", "Headphones";
++
++		simple-audio-card,dai-link@1 {
++			format = "i2s";
++
++			cpu {
++				sound-dai = <&i2s_8ch_2>;
++			};
++
++			codec {
++				sound-dai = <&codec>;
++			};
++
++		};
++
++	};
++
++	pcm5102_sound: pcm5102-sound {
++		compatible = "simple-audio-card";
++		simple-audio-card,format = "i2s";
++		simple-audio-card,mclk-fs = <256>;
++		simple-audio-card,name = "pcm5102a";
++
++		simple-audio-card,dai-link@1 {
++			format = "i2s";
++			cpu {
++				sound-dai = <&i2s_8ch_0>;
++			};
++
++			codec {
++				sound-dai = <&pcm5102a>;
++			};
++		};
++	};
++
++	pcm5102a: pcm5102a {
++			#sound-dai-cells = <0>;
++			compatible = "ti,pcm5102a";
++			pcm510x,format = "i2s";
++	};
++
+ 	sdio_pwrseq: sdio-pwrseq {
+ 		compatible = "mmc-pwrseq-simple";
+ 		pinctrl-0 = <&wifi_enable_h>;
+@@ -128,6 +176,11 @@ vdd_log: vdd-log {
+ 	};
+ };
+ 
++&codec {
++	status = "okay";
++	#sound-dai-cells = <0>;
++};
++
+ &cpu0 {
+ 	cpu-supply = <&vdd_core>;
+ };
+@@ -139,6 +192,7 @@ &emmc {
+ 	non-removable;
+ 	pinctrl-names = "default";
+ 	pinctrl-0 = <&emmc_bus8 &emmc_clk &emmc_cmd>;
++	mmc-hs200-1_8v;
+ 	vmmc-supply = <&vcc_io>;
+ 	status = "okay";
+ };
+@@ -242,6 +296,19 @@ &io_domains {
+ 	status = "okay";
+ };
+ 
++&i2s_8ch_0 {
++	#sound-dai-cells = <0>;
++	assigned-clocks = <&cru SCLK_I2S0_8CH_RX>;
++	assigned-clock-parents = <&cru SCLK_I2S0_8CH_TX_MUX>;
++	rockchip,clk-trcm = <1>;
++	status = "okay";
++};
++
++&i2s_8ch_2 {
++	#sound-dai-cells = <0>;
++	status = "okay";
++};
++
+ &pinctrl {
+ 	pinctrl-names = "default";
+ 	pinctrl-0 = <&rtc_32k>;
+@@ -333,10 +400,18 @@ &sdmmc {
+ 	cap-mmc-highspeed;
+ 	cap-sd-highspeed;
+ 	disable-wp;
++	pinctrl-0 = <&sdmmc_clk &sdmmc_cmd &sdmmc_det &sdmmc_bus4>;
++	card-detect-delay = <800>;
```

Comment:
This should already match the default `pinctrl-0` in `rk3308.dtsi`, not need to override here.

Is the `card-detect-delay` needed on your boards/sd-cards?

Reply:
Good catch.  The pinctrl-0 should be removed from the board rk3308-rock-pi-s.dts
The 800ms delay is probably longer than needed, but why mess with it?
There may be some strange SDcard out there that take a long time to initialize on power up.
Will you really be bothered by waiting another 500ms after inserting each SDcard?

Reply:
In other news:  The 16MB CMA is causing performance to lag.
Specifically, I notice that apt package installs are slower.
Checking with /proc/meminfo shows that >32MB are used if available.
I think we should up the CMA size to 64MB.  OK with you?

Reply:
@Kwiboo the `card-detect-delay` was present in the original dtb 4.4 kernel from rockchip, altough with a typo (the property was written as `ard-detect-delay = <0x320>`) that made it uneffictive. 

Anyway, I "rebased" upon the previous @brentr 's patch that was disabled during a kernel bump due to being unable to apply correctly anymore (see https://github.com/armbian/build/blob/main/patch/kernel/archive/rockchip64-6.12/board-rockpis-0001-arm64-dts.patch.disabled#L256-L257)

No problem in removing the property though, if it is redundant.